### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by wesho1107**
Update User Guide and Developer Guide with necessary information such as UML Class Diagrams

### CODE REVIEW

**Review by Respirayson — APPROVED** [review]
LGTM

**Review by raydenlim — COMMENTED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -420,6 +430,8 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 | `* * *`  | busy avenger                               | keep track of what needs to be done              | better guide my students.                                              |
 | `* * *`  | conscientious avenger                      | view my students' grades for their assignments   | better assess my students' competency.                                 |
 | `* * *`  | caring avenger                             | view my students' comments for their assignments | take note of my students' strengths and weaknesses.                    |
+| `* * *`  | user                                       | create a new consultation with students          | keep track of when and with who the consultation is held               |
+| `* * *`  | user                                       | add students to an existing consultation         | invite more students to join a consultation discussion                 |
```
Comment (Originalidk) [inline docs/DeveloperGuide.md:434]: Might want to add a characteristic to the user

**Review by Originalidk — APPROVED** [review]
LGTM, just a very small nit

**Pull request merged**